internal/envoy/tcp_keepalive.go

```diff
@@ -0,0 +1,71 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package envoy
+
+import (
+	"syscall"
+
+	envoy_api_v2_core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+)
+
+// We only support Envoy on Linux so always configure Linux TCP keep-alive
+// socket options regardless of the platform that Contour is running on.
+const (
+	TCP_KEEPIDLE  = 0x4 // Linux syscall.TCP_KEEPIDLE
+	TCP_KEEPINTVL = 0x5 // Linux syscall.TCP_KEEPINTVL
+	TCP_KEEPCNT   = 0x6 // Linux syscall.TCP_KEEPCNT
+)
+
+func TCPKeepaliveSocketOptions() []*envoy_api_v2_core.SocketOption {
```

Comment:
@erwbgy we should document this (could be this PR or a new one) to make the defaults here known. Also, we should think about adding these to the config file to allow users to define their own timeouts.

Reply:
Indeed - added a suggestion about documenting this change to #2633 and would like to do docs and configuration in separate PRs, if that's okay.

Reply:
Added documentation for Envoy TCP Keepalives.